apps/aecore/lib/aecore/persistence/worker.ex

```diff
@@ -30,27 +31,56 @@ defmodule Aecore.Persistence.Worker do
   end
   def read_block_by_hash(_hash), do: {:error, "bad hash value"}
 
+  @spec write_chain_state_by_pubkey(chain_state :: map()) :: :ok | {:error, reason :: term()}
+  def write_chain_state_by_pubkey(%{} = chain_state) do
+    {:ok, pubkey} = Keys.pubkey()
+    GenServer.call(__MODULE__, {:write_chain_state_by_pubkey, {pubkey, chain_state}})
+  end
+  def write_chain_state_by_pubkey(_chain_state), do: {:error, "bad chain_state structure"}
+
+  @spec read_chain_state_by_pubkey() ::
+  {:ok, chain_state :: map()} | :not_found | {:error, reason :: term()}
+  def read_chain_state_by_pubkey() do
+    {:ok, pubkey} = Keys.pubkey()
+    GenServer.call(__MODULE__, {:read_chain_state_by_pubkey, pubkey})
+  end
+
   ## Server side
 
   def init(_) do
     ## We are creating `blocks_by_hash` family for the blocks
+    ## and `chain_state_by_pubkey` as well
     ## https://github.com/facebook/rocksdb/wiki/Column-Families
-    {:ok, db, %{"blocks_by_hash" => blocks_family}} =
+    {:ok, db, %{"blocks_by_hash"       => blocks_family,
+                "chain_state_by_pubkey" => chain_state_family}} =
       Rox.open(persistence_path(),
-        [create_if_missing: true, auto_create_column_families: true], ["blocks_by_hash"])
-    {:ok, %{db: db, blocks_family: blocks_family}}
+        [create_if_missing: true, auto_create_column_families: true],
+        ["blocks_by_hash", "chain_state_by_pubkey"])
+    {:ok, %{db: db,
+            blocks_family: blocks_family,
+            chain_state_family: chain_state_family}}
   end
 
   def handle_call({:write_block_by_hash, {hash, block}}, _from,
     %{db: _db, blocks_family: blocks_family} = state) do
     {:reply, Rox.put(blocks_family, hash, block), state}
   end
 
+  def handle_call({:write_chain_state_by_pubkey, {pubkey, chain_state}}, _from,
+    %{db: _db, chain_state_family: chain_state_family} = state) do
+    {:reply, Rox.put(chain_state_family, pubkey, chain_state), state}
```

Comment:
does this write every public key as seperate entry in the db?

Reply:
yes

Reply:
great!